### ISSUE TITLE
generate stats from manual filed statbook

### ISSUE THREAD

**Issue by Ganelous04** [dhmay/jamstats#132]
Hi,
Is it possible to use jamstats from a statbook filed manually?
I filed a xslx statbook manually (IGRF, Score, Penalties and lineup sheets), imported it in CRG using "Import statbook" button in _Manage data_ page, then exported the json but when I upload it on jamstats in line, it returns an error "Error loading game file. Please check that the file is a valid JSON file."
The json file seems to be ok (there is data in it).

When I try using jamstat.exe the folowing message occures :
```
Plotting plot_jammer_stats_team1
Failed to make skater plot plot_jammer_stats_team1: min() arg is an empty sequence
Plotting plot_jammer_stats_team2
Failed to make skater plot plot_jammer_stats_team2: min() arg is an empty sequence
Plotting plot_skater_stats_team1
Failed to make skater penalty subplot:
Traceback (most recent call last):
  File "jamstats\plots\skaterplots.py", line 155, in plot_skater_stats
IndexError: list index out of range

Failed to make skater plot plot_skater_stats_team1: min() arg is an empty sequence
Plotting plot_skater_stats_team2
Failed to make skater penalty subplot:
Traceback (most recent call last):
  File "jamstats\plots\skaterplots.py", line 155, in plot_skater_stats
IndexError: list index out of range

Failed to make skater plot plot_skater_stats_team2: min() arg is an empty sequence
[13228] Failed to execute script 'jamstats' due to unhandled exception!
Traceback (most recent call last):
  File "jamstats", line 241, in <module>
  File "jamstats", line 237, in __main__
  File "jamstats\plots\plot_together.py", line 50, in save_game_plots_to_pdf
  File "jamstats\plots\plot_together.py", line 104, in make_all_plots
ValueError: max() arg is an empty sequence
```

I use CRG 5.0.11.

Thanks for you help

**Comment by bullseye555** [comment]
Ganelous04

This will be because CRG only imports the relevant sections of a statsbook
to start a game - a completed stats book will not import into CRG with the
Jam by Jam data and then export as a fully filled JSON

What you're trying to do isn't possible with CRG - Damon would have to
build Jamstats to import from a Statsbook XLSX to get the data

**Comment by dhmay** [comment]
Yep, that's right, but if you'd like to be able to run Jamstats on this kind of JSON file, then I can see if I can support it. I'd rather try to support the JSON export than the XLSX... could you please upload that JSON file, and I'll see how tough that would be?

**Comment by dhmay** [comment]
Thanks! I'm out of town right now, will have a look next week.

**Comment by bullseye555** [comment]
Looking at the JSON from @Ganelous04, my suspicion is correct.
CRG has only imported the IGRF page data. None of the Lineups, Penalties, or Score have been imported

The key clue to this prior to even be ing exported is that if you were to open up the SBO Page and click 'Start New Game', you'd be able to select this game (see JSON screenshot below that shows the game status as "Prepared") 
![Screenshot_20230416-101541](https://user-images.githubusercontent.com/28804763/232259554-6149783f-eb82-4fa2-ba50-82fc4301d874.png)

In short, there's nothing in the JSON file that @dhmay will be able to use. It'll have to be from the XLSX file

**Comment by dhmay** [comment]
Yep, confirmed. I stepped through the whole file. There appears to be full roster data in there, but that's it. There's a single dummy jam, with no data (0 seconds, 0 points, no skater indicated as jammer/pivot). This file just doesn't contain any game data, unfortunately. As @bullseye555 said, this looks like a JSON file that's ready to *start* a game with.

Which seems weird to me. So, @Ganelous04 , you created this file by importing a completed statbook into CRG and then exporting the game as json, right? When you've loaded it into CRG, do you see all the penalties you entered showing up in the penalty sheet in CRG?

I don't know anything about how this works (never dealt with statbooks), but if it's been imported into CRG successfully I wouldn't think it would matter whether the penalty data were generated in CRG or in xlsx.

**Comment by bullseye555** [comment]
For context, CRG only imports the IGRF page of the StatsBook, it doesn't even look at the other pages, so that data will never be imported and available in CRG

(prior to JamStats existing, there as 0 reason to upload a complete game Statsbook into CRG AS a completed game)

**Comment by dhmay** [comment]
Ah. Well, sounds like there's nothing to be done, then.
